core/domain/exp_services.py

```diff
@@ -792,6 +793,15 @@ def _save_exploration(committer_id, exploration, commit_message, change_list):
                 'which is too old. Please reload the page and try again.'
                 % (exploration_model.version, exploration.version))
 
+    if feconf.ENABLE_STRING_CLASSIFIER:
```

Comment:
@seanlip Because we wanted to pass the correct version of the exploration to the functions that handle job creations. The get_trainable_states() function needs the old exploration domain object. The job/mapping creation requires the new updated exploration domain object (with incremented version).